src/content/language/dot-shorthand.md

```diff
@@ -0,0 +1,328 @@
+:::
+
+Write more concise Dart code by using the dot shorthand syntax `.foo` 
+instead of `ContextType.foo`. The compiler can infer the type 
+whenever it's clear from the context, letting you access 
+enum values, static members, or constructors more cleanly.
+
+Here is a quick look:
+
+<?code-excerpt "language/lib/classes/shorthand.dart (intro)"?>
+```dart
+// Enum example
+enum Status { none, running, stopped, paused }
+Status currentStatus = .running; // Instead of Status.running
+
+// Static method example
+int port = .parse('8080'); // Instead of int.parse('8080')
+
+// Constructor example
+class Point {
+  final int x, y;
+  Point(this.x, this.y);
+  Point.origin() : x = 0, y = 0;
+}
+Point origin = .origin(); // Instead of Point.origin()
+```
+
+## The role of context type 
+Dot shorthand syntax works because the compiler understands 
+the context type. This is the type it expects an expression 
+to have based on its location. For example, 
+in `Status currentStatus = .running`, the compiler knows 
+a `Status` is expected, so it infers `.running` to mean 
+`Status.running`.
+
+## Using dot shorthand syntax
+
+### Enums
+A primary and highly recommended use case for dot shorthands 
+is with enums, especially in assignments and switch statements, 
+where the enum type is very obvious.
+
+<?code-excerpt "language/lib/classes/shorthand.dart (enums)"?>
+```dart
+// Enum example
+enum LogLevel { debug, info, warning, error }
+
+// Function to get a color code based on log level
+String colorCode(LogLevel level) {
+  return switch (level) {
+    .debug   => 'gray',    // Instead of LogLevel.debug
+    .info    => 'blue',    // Instead of LogLevel.info
+    .warning => 'orange',  // Instead of LogLevel.warning
+    .error   => 'red',     // Instead of LogLevel.error
+  };
+}
+
+// Example usage:
+var warnColor = colorCode(.warning); // Returns 'orange'
+```
+
+### Named constructors
+Dot shorthands are useful for invoking named constructors, 
+static factory methods, or constructors of generic classes 
+where type arguments are provided.
+
+<?code-excerpt "language/lib/classes/shorthand.dart (constructors)"?>
+```dart
+ class Point {
+  final double x, y;
+  const Point(this.x, this.y);
+  const Point.origin() : x = 0, y = 0; // Named constructor
+  static Point fromList(List<double> list) => Point(list[0], list[1]); // Static method
+}
+
+// Named constructor
+Point origin = .origin(); // Instead of Point.origin()
+
+// Static factory method
+Point p1 = .fromList([1.0, 2.0]); // Instead of Point.fromList([1.0, 2.0])
+
+// With type arguments for generic class constructors
+List<int> intList = .filled(5, 0); // Instead of List.filled(5, 0)
+``` 
+### Unnamed constructors
+The `.new` dot shorthand provides a concise way to call an
+unnamed constructor of a class. This is useful 
+for instantiating fields and variables where the type is 
+already explicitly declared.
+
+<?code-excerpt "language/lib/classes/shorthand.dart (unnamed)"?>
+```dart
```

Comment:
Maybe this code snippet would be easier to read if you broke it out into two code blocks (one showing before and one showing after) rather than having it commented as before and after in a single code block.